### PR TITLE
adding repository -> rule id -> timing info

### PR DESCRIPTION
**Pull request by colleend**
* now the `output_per_rule_json_results` leads to the creation of a json file with repository -> rule_id timing information 

_Test Plan:_
Run benchmarks with the command:
`./run-benchmarks --output-time-per-rule-json <file name>`
and check that the file created has rule timing information.

### CODE REVIEW

File: perf/RepositoryTimePerRule.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self, 
+        output_file: str,
+        repo_to_times_per_rule: Dict[str, Dict[str, int]] = defaultdict(dict)
+    ) -> None:
+        self.repo_to_times_per_rule = repo_to_times_per_rule
+        self.output_file = output_file
+
+    def times_per_file_to_times_per_rule(self, repo_name: str, times_per_file_bytes: bytes):
+        try:
+            times_per_file = json.loads(times_per_file_bytes.decode('utf-8'))
```
[the remark this answers is not in the document]
Comment (colleend) [reply perf/RepositoryTimePerRule.py:20]: handling the unicodedecode error, so this isn't a problem

[the remark this answers is not in the document]
Comment (mschwager) [reply perf/RepositoryTimePerRule.py:20]: We should improve this rule to consider this case :+1:

[the remark this answers is not in the document]
Comment (mschwager) [reply perf/RepositoryTimePerRule.py:20]: I think we need this first: https://github.com/returntocorp/semgrep/issues/3159

[the remark this answers is not in the document]
Comment (colleend) [reply perf/RepositoryTimePerRule.py:20]: ah yes good point!!

**Review by mschwager — APPROVED** [review]

**Pull request merged**